src/ai/backend/common/exception.py

```diff
@@ -59,7 +59,10 @@ class InvalidImageTag(ValueError):
     """
 
     def __str__(self) -> str:
-        return f"Invalid or duplicate image name tag: {self.args[0]}"
+        if len(self.args) > 1 and self.args[1] is not None:
+            return f"Invalid or duplicate image name tag: {self.args[0]}, full image name: {self.args[1]}"
+        else:
+            return f"Invalid or duplicate image name tag: {self.args[0]}"
```

Comment:
Instead of calculating `len(self.args)` like this, we could consider adding custom attributes to the exception class in a way to be safe with pickling and unpickling.

Check out the discussions here: https://stackoverflow.com/a/27334007/122423